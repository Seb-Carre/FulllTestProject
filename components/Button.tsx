import {Pressable, PressableProps, StyleSheet} from 'react-native';

import CustomText from './CustomText';
import React from 'react';

interface ButtonProps extends PressableProps {
  children: any;
}

export default function Button({children, ...props}: ButtonProps) {
  return (
    <Pressable style={styles.container} {...props}>
      <CustomText style={styles.buttonText}>{children}</CustomText>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#00a2ff',
    paddingHorizontal: 50,
    paddingVertical: 20,
    alignItems: 'center',
    borderRadius: 10,
  },
  buttonText: {
    fontSize: 20,
  },
});
