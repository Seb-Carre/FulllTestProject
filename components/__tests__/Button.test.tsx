import {fireEvent, render} from '@testing-library/react-native';

import Button from '../Button';
import React from 'react';

it('render the button as the snapShot we expected', () => {
  const {toJSON} = render(<Button>Hello World</Button>);

  expect(toJSON()).toMatchSnapshot();
});

it('renders the button with the text correctly and click', () => {
  const text = 'Click on me';
  // Using a mock function to check if our function is called
  const mockFunction = jest.fn();
  const {getByTestId, queryByText} = render(
    <Button testID="button" onPress={mockFunction}>
      {text}
    </Button>,
  );
  const button = getByTestId('button');
  // We check if the text button is here
  expect(queryByText(text)).not.toBeNull();

  // Making a press event on our button
  fireEvent.press(button);
  // With the press, we know that our function has been called !
  expect(mockFunction).toBeCalled();
});
