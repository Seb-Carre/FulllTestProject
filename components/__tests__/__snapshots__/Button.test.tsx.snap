// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the button as the snapShot we expected 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#00a2ff",
      "borderRadius": 10,
      "paddingHorizontal": 50,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      [
        {},
        {
          "fontSize": 20,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
